### DISCUSSION TITLE
analog audio in convert to pwm

### DISCUSSION THREAD

**Discussion by Simonthorsted** [arendst/Tasmota#19828] [Q&A]
Hello is an esp fast enough to convert analog audio in(pin) and convert to pwm out to a servo, to make a skull move it`s jaw like
it is talking ,if yes do some one know or can make a rule please .

**Comment by s-hadinger** [comment]
Oh this looks fun. Yes it's definitely possible but on esp32 or any variant (not esp8266)

You will need an I2S mems microphone

**Reply by Simonthorsted** [reply]
i have 10 of this in order from aliexpress I2S Interface INMP441to make wled blink to music to ,so when they come i can go to work on making the jaw thing to then.

**Reply by Simonthorsted** [reply]
is it then a rule that makes the convert ,do you know how ?

**Comment by Simonthorsted** [comment]
Hello thanks for your replay Do you know if ,it is a rule that I need ,to make the pwm output to a motor to make the jaw move or ????
-------- Oprindelig besked --------Fra: s-hadinger ***@***.***> Dato: 23.10.2023  21.24  (GMT+01:00) Til: arendst/Tasmota ***@***.***> Cc: simon thorsted ***@***.***>, Author ***@***.***> Emne: Re: [arendst/Tasmota] analog audio in convert to pwm (Discussion #19828) 
Oh this looks fun. Yes it's definitely possible but on esp32 or any variant (not esp8266)
You will need an I2S mems microphone

**Reply by sfromis** [reply]
Probably not a rule, but some coding, like using Berry.
